app/static/app/js/components/Paginated.jsx

```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+import update from 'react-addons-update';
+
+/*abstract*/ class Paginated extends React.Component{
+  constructor(){
+    super();
+  }
+
+  updatePagination(itemsPerPage, totalItems){
+    let currentPage = 1;
+    const totalPages = this.totalPages(itemsPerPage, totalItems);
+
+    if (this.state.pagination && this.state.pagination.currentPage !== undefined){
```

Comment:
I tend to use null to indicate a variable that has been initialized to "nothing", while undefined for a variable that has not even been initialized. In this case my reason was to check whether pagination.currentPage has been initialized. Both approaches are fine and loose equality here would not not harm.